libusb/os/events_posix.c

```diff
@@ -224,16 +224,15 @@ int usbi_alloc_event_data(struct libusb_context *ctx)
 	struct pollfd *fds;
 	size_t i = 0;
 
-	if (ctx->event_data) {
-		free(ctx->event_data);
-		ctx->event_data = NULL;
-	}
-
+	free(ctx->event_data);
+	ctx->event_data = NULL;
 	ctx->event_data_cnt = 0;
+
+	unsigned int cnt = 0;
```

Comment:
This should be at the top.

Reply:
Do we still need to support C89?

Reply:
It is about consistency with the surrounding code. I am personally a fan of limiting scope for variables and declaring them when needed, but that is not the question here. It just lookw weird with some variables declared at the top and this single one in the middle.

Reply:
I am also a fan of limiting scope and locality of reference.  How about I just rehabilitate the function then?  See new commit.

Also, clang static analyzer was warning in this function (both before and after my changes) `Use of memory allocated with size zero`. Not sure this is a true positive or false positive (the `for_each_event_source` to too gnarly for me the try and understand right now). I added an early return is cnt is zero. That seem correct?